react/index.tsx

```diff
@@ -56,6 +65,19 @@ const AddressChallenge: StorefrontFunctionComponent<WrappedComponentProps &
     [logisticsData?.logistics?.googleMapsKey]
   )
 
+  const updateRegionID = async () => {
```

Comment:
You could do something like this (not sure if the Address type is defined, but you get the idea):
```suggestion
  const updateRegionID = async (address: Address) => {
```
and then use the passed-in address to fill the `setRegionId` mutation variables. You might not even need to remove the validation.

Reply:
What if instead of using the typing and `addressFields` we could use the address from the `orderForm` using the `Address` query used up top?

Reply:
I don't think that's necessary. You already have the new address since it was used to send the `updateAddress` mutation. If you were going to use the `orderForm` you would need to make sure to refetch it after sending the `updateAddress` mutation so that you have the newly updated address, and there might be a delay there. Better to use the same address that the mutation used, in my opinion.

Reply:
I updated the update region Address Passing to it only the necessary option used in the `updateRegion` function.